powerauth-restful-security-spring/src/main/java/io/getlime/security/powerauth/rest/api/spring/filter/EncryptionResponseBodyAdvice.java

```diff
@@ -0,0 +1,95 @@
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package io.getlime.security.powerauth.rest.api.spring.filter;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.getlime.security.powerauth.crypto.lib.encryptor.ecies.EciesDecryptor;
+import io.getlime.security.powerauth.crypto.lib.encryptor.ecies.model.EciesCryptogram;
+import io.getlime.security.powerauth.rest.api.base.encryption.PowerAuthEciesEncryption;
+import io.getlime.security.powerauth.rest.api.spring.annotation.PowerAuthEncryption;
+import org.springframework.core.MethodParameter;
+import org.springframework.http.MediaType;
+import org.springframework.http.converter.HttpMessageConverter;
+import org.springframework.http.server.ServerHttpRequest;
+import org.springframework.http.server.ServerHttpResponse;
+import org.springframework.http.server.ServletServerHttpRequest;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseBodyAdvice;
+
+import javax.servlet.http.HttpServletRequest;
+import java.io.ByteArrayOutputStream;
+
+/**
+ * Controller advice used for encryption of responses of REST endpoints.
+ *
+ * @author Roman Strobl, roman.strobl@wultra.com
+ */
+@ControllerAdvice
+public class EncryptionResponseBodyAdvice implements ResponseBodyAdvice<Object> {
+
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    /**
+     * Whether method supports encryption.
+     *
+     * @param methodParameter Method parameter.
+     * @param aClass HTTP message converter.
+     * @return Whether method supports encryption.
+     */
+    @Override
+    public boolean supports(MethodParameter methodParameter, Class<? extends HttpMessageConverter<?>> aClass) {
+        return methodParameter.hasMethodAnnotation(PowerAuthEncryption.class);
+    }
+
+    /**
+     * Encrypt response before writing body.
+     *
+     * @param response Response object.
+     * @param methodParameter Method parameter.
+     * @param mediaType Media type.
+     * @param aClass HTTP message converter.
+     * @param serverHttpRequest HTTP request.
+     * @param serverHttpResponse HTTP response.
+     * @return ECIES cryptogram.
+     */
+    @Override
+    public EciesCryptogram beforeBodyWrite(Object response, MethodParameter methodParameter, MediaType mediaType, Class<? extends HttpMessageConverter<?>> aClass, ServerHttpRequest serverHttpRequest, ServerHttpResponse serverHttpResponse) {
```

Comment:
There will be a small change in this code in next pull request, I introduce model classes for ECIES requests and responses which use Strings as parameters and are used in all ECIES requests/responses to restful-integration. I rely here on default Jackson's serialization of byte[] to Base64 encoding, however I think it is better if we handle the conversion to Base64 ourselves.